warehouse/query-core/src/main/java/datawave/query/tables/BatchScannerSession.java

```diff
@@ -541,10 +542,22 @@ public void onSuccess(Scan finishedScan) {
      */
     @Override
     public void onFailure(Throwable t) {
-        stop();
+        if (isInterruptedException(t)) {
+            log.info("BatchScannerSession interrupted");
+        } else {
+            log.error("BatchScanerSession failed", t);
+        }
         uncaughtExceptionHandler.uncaughtException(Thread.currentThread().currentThread(), t);
+        stopAsync();
         Throwables.propagate(t);
-        
+    }
+    
+    private boolean isInterruptedException(Throwable t) {
```

Comment:
Would it be worthwhile to introduce commons lang dependency on ExceptionUtils for this instead?

Reply:
I will check it out

Reply:
Actually the exception is not always the root cause, so I think I am going to stick with what I have.